package/MDAnalysis/core/Selection.py

```diff
@@ -30,105 +35,100 @@
 from numpy.lib.utils import deprecate
 from Bio.KDTree import KDTree
 import warnings
+import logging
 
-from .AtomGroup import AtomGroup, Universe
 from MDAnalysis.core import flags
 from ..lib import distances
 from ..lib.mdamath import triclinic_vectors
+from ..exceptions import SelectionError, NoDataError
 
 
-class Selection(object):
```

Comment:
I never understood the caching thing, as far as I could tell each Selection object was a single use thing, so caching was a waste.